### PR TITLE
added support for info_overlay and infodecal

### PR DESCRIPTION
**Pull request by half5life**

### CODE REVIEW

File: source1/bsp/entities/base_entity_handler.py [hunk 1]
```diff
@@ -659,13 +659,42 @@ def handle_env_sun(self, entity: env_sun, entity_raw: dict):
...
+        size = bpy.data.images[tex_name].size
+
+        xCor = size[0]/8
+        zCor = size[1]/8
+        verts = [
+            [-xCor, 0, -zCor],
+            [xCor, 0, -zCor],
+            [xCor, 0, zCor],
+            [-xCor, 0, zCor]
+            ]
```
Comment (REDxEYE) [inline source1/bsp/entities/base_entity_handler.py:684]: Please, use snake_case variable names

File: source1/bsp/import_bsp.py [hunk 2]
```diff
@@ -280,6 +282,123 @@ def load_disp(self):
...
+        info_overlay_lump: Optional[OverlayLump] = self.map_file.get_lump('LUMP_OVERLAYS')
+        if info_overlay_lump:
+            parent_collection = get_or_create_collection('info_overlays', self.main_collection)
+            overlays = info_overlay_lump.overlays
+            for overlay in overlays:
+                verts = np.array(overlay.UVPoints).reshape((4,3))
+                for vert in verts:
+                    temp = vert[1]
+                    vert[1] = vert[2]
+                    vert[2] = temp
```
Comment (REDxEYE) [inline source1/bsp/import_bsp.py:295]: Can be replaced with a bit more readable 
`vert[1], vert[2] = vert[2], vert[1]`

File: source1/bsp/import_bsp.py [hunk 3]
```diff
@@ -280,6 +282,123 @@ def load_disp(self):
...
+                uv_data[3].uv[1] = min(overlay.V) #min
+
+                bpy.ops.object.select_all(action='DESELECT')
+                mesh_obj.select_set(True)
+                bpy.context.view_layer.objects.active = mesh_obj
+                bpy.ops.object.mode_set(mode='EDIT')
+                bpy.ops.mesh.select_all(action='SELECT')
+
+                bpy.ops.mesh.flip_normals()
+                bpy.ops.object.mode_set()
```
Comment (REDxEYE) [inline source1/bsp/import_bsp.py:327]: mesh_data.flip_normals()

File: source1/bsp/import_bsp.py [hunk 4]
```diff
@@ -280,6 +282,123 @@ def load_disp(self):
...
+                mesh_obj.rotation_quaternion = thisRot
+
+                for pos in range(0,2):
+                    for i in range(0,3):
+                        mesh_obj.location[i] = mesh_obj.location[i] + (self.scale*pos*thisNorm[i])
+                
+                bpy.ops.object.transform_apply(location=False, rotation=True, scale=False)
+                mesh_obj.select_set(False)
+    
+    def rotate_infodecals(self):
```
Comment (REDxEYE) [inline source1/bsp/import_bsp.py:354]: add an underscore to utility function name

File: source1/bsp/import_bsp.py [hunk 5]
```diff
@@ -280,6 +282,123 @@ def load_disp(self):
...
+                calc_distance = Vector(obj.location - func.location).length
+                if calc_distance<=func_distnace:
+                    func_distnace = calc_distance
+                    func_name = func.name
+                
+            func_obj = bpy.data.objects[func_name]
+            directions = [[1,0,0],[0,1,0],[0,0,1],[-1,0,0],[0,-1,0],[0,0,-1]]
+            rays_hits = []
+            for dirF in directions:
+                rayF = func.ray_cast(obj.location, (dirF[0], dirF[1], dirF[2]))
```
Comment (REDxEYE) [inline source1/bsp/import_bsp.py:369]: snake_case variable names

File: source1/bsp/import_bsp.py [hunk 6]
```diff
@@ -280,6 +282,123 @@ def load_disp(self):
...
+                if (rayF[0]):
```
Comment (REDxEYE) [inline source1/bsp/import_bsp.py:370]: redundant parentheses

File: source1/bsp/lumps/overlay_lump.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    def __init__(self, bsp, lump_id):
+        super().__init__(bsp, lump_id)
+        self.overlays = []
+    
+    def parse(self):
+        reader = self.reader
+        while reader:
+            overlay = Overlay(self, self._bsp).parse(reader)
+            self.overlays.append(overlay)
+        return self
```
Comment (REDxEYE) [inline source1/bsp/lumps/overlay_lump.py:18]: Add new line here

File: source1_operators.py [hunk 8]
```diff
@@ -95,6 +95,8 @@ def execute(self, context):
         bsp_map.load_disp()
         bsp_map.load_entities()
         bsp_map.load_static_props()
+        bsp_map.load_overlays()
+        bsp_map.rotate_infodecals()
```
Comment (REDxEYE) [inline source1_operators.py:99]: Move this function call into load_overlays method

**Review by REDxEYE — CHANGES_REQUESTED** [review]
Please fix all mentioned issues and I happily merge your PR

**Comment by ShadelessFox** [comment]
Consider checking for the existence of requested material:

```
[INFO    ]-[decal.bsp] Loading map "C:\Users\ShadelessFox\Desktop\decal.bsp"
[DEBUG   ]-[decal.bsp] Adding map pack file to content manager
[INFO    ]-[content_manager] Requesting materials\dev\dev_measurecrate01.vmt file
Python: Traceback (most recent call last):
  File "E:\SteamLibrary\steamapps\common\Blender\2.92\scripts\addons\SourceIO\source1_operators.py", line 96, in execute
    bsp_map.load_entities()
  File "E:\SteamLibrary\steamapps\common\Blender\2.92\scripts\addons\SourceIO\source1\bsp\import_bsp.py", line 92, in load_entities
    self.entity_handler.load_entities()
  File "E:\SteamLibrary\steamapps\common\Blender\2.92\scripts\addons\SourceIO\source1\bsp\entities\base_entity_handler.py", line 298, in load_entities
    if not self.handle_entity(entity_data):
  File "E:\SteamLibrary\steamapps\common\Blender\2.92\scripts\addons\SourceIO\source1\bsp\entities\base_entity_handler.py", line 312, in handle_entity
    handler_function(entity_object, entity_data)
  File "E:\SteamLibrary\steamapps\common\Blender\2.92\scripts\addons\SourceIO\source1\bsp\entities\base_entity_handler.py", line 674, in handle_infodecal
    tex_name = bpy.data.materials[material_name].node_tree.nodes["$basetexture"].image.name
KeyError: 'bpy_prop_collection[key]: key "dev_measurecrate01" not found'
```

**Comment by ShadelessFox** [comment]
```
OrderedDict([('origin', '-384 128 273'),
             ('angles', '0 0 0'),
             ('classname', 'info_player_start'),
             ('hammerid', 156)])
GLump "sprp" offset: 36 size: 12
GLump "dprp" offset: 48 size: 12
Python: Traceback (most recent call last):
  File "E:\SteamLibrary\steamapps\common\Blender\2.92\scripts\addons\SourceIO\source1_operators.py", line 99, in execute
    bsp_map.rotate_infodecals()
  File "E:\SteamLibrary\steamapps\common\Blender\2.92\scripts\addons\SourceIO\source1\bsp\import_bsp.py", line 359, in rotate_infodecals
    for func in bpy.data.collections['func_brush'].all_objects:
KeyError: 'bpy_prop_collection[key]: key "func_brush" not found'
```

The game used for testing is Half-Life 2: [decal.zip](https://github.com/REDxEYE/SourceIO/files/6095882/decal.zip)

**Review by REDxEYE — APPROVED** [review]
Much better now. Let's do some testing and if everything will be stable, I'll merge it into master.
For now, I create an experimental release with this feature

**Pull request merged**